draft-pda-protocol.md

```diff
@@ -529,81 +527,69 @@ example, that clients are prohibited from talking to helpers but not the leader.
 Is it OK that leaders learn that about a client? I'm not sure, so I'd be
 inclined to remove this unless we have a concrete use case.]
 
-Next, the client issues a POST request to `[leader]/upload_finish`. It begins by
-setting up an HPKE {{!I-D.irtf-cfrg-hpke}} context for the helper by running
-
-~~~
-enc, context = SetupBaseS(pk, [TODO])
-~~~
-
-where `pk` is the KEM public key encoded by `HpkeConfig.public_key`. The outputs
-are the helper's encapsulated context `enc` and the context `context`.
-
-Next, the client encodes its measurements as an input for the PA protocol. It
-then generates a validity proof for its input and uses `context` to split the
-input and proof into a *leader share* and a *helper share*, where the latter is
-protected by the HPKE context. Note that the details of each of these processing
-steps --- encode, prove, split, and encrypt --- are specific to the PA protocol.
-
-[TODO: Fully specify encryption of the helper's share. We need to make sure we
-authenticate things like the PA parameters and the report timestamp. We need to
-decide what the info string for SetupBaseS() will be, as well as the aad for
-context.Seal(). The aad might be the entire "transcript" between the client and
-helper.]
-
-[OPEN ISSUE: Is it safe to generate the proof once, then secret-share between
-each (leader, helper) pair? Probably not in general, but maybe for Prio?]
-
-[OPEN ISSUE: allow server to send joint randomness in UploadStartResp, and then
-enforce uniqueness via double-spend state or something else (see issue#48).]
-
-The payload of the POST request to `[leader]/upload_finish` is structured as
-follows:
+Next, the client issues a POST request to `[leader]/upload_finish`. The payload
+is structured as follows:
 
 ~~~
 struct {
-  PATask task;                 // Equal to PAUploadStartReq.task
-  uint64 time;                 // UNIX time (in seconds).
-  uint8 helper_hpke_config_id; // Equal to HpkeConfig.id
-  PAHelperInputShare helper_input_share;
-  PALeaderInputShare leader_input_share;
+  PATask task; // Equal to PAUploadStartReq.task
+  uint64 time; // UNIX time (in seconds).
+  PAEncryptedInputShare encrypted_input_shares<1..2^16-1>;
```

Comment:
nit: isn't this 7..2^16-1:

1 for config id
2 each for the lengths of enc and payload
1 each for the minimum size of length and payload.

Reply:
Yeah, well, we'll need to fix up the bounds elsewhere as well. I'll add a TODO to handle this later, once the protocol gets closer to settled.